PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskIDFragmentationFunction.cxx

```diff
@@ -2039,9 +2033,7 @@ AliEmcalJet* AliAnalysisTaskIDFragmentationFunction::GetPerpendicularCone(AliEmc
   if (!processedJet)
     return 0x0;
   
-  Double_t dDistance = 2.0 * TMath::Abs(GetFFRadius());
-  
-  AliEmcalJet* perpJet = new AliEmcalJet(processedJet->Pt(), processedJet->Eta(), processedJet->Phi()+perpAngle, processedJet->M());
+  AliEmcalJet* perpJet = new AliEmcalJet(processedJet->Pt(), processedJet->Eta(), processedJet->Phi() + perpAngle, processedJet->M());
```

Comment:
In case this is just used locally inside the function it should be better put on the stack (AliEmcalJet perpJet(...);) in order to avoid memory leaks.

Reply:
No, they are returned and saved in a list to be processed later. The list is the owner of the stored objects and is properly deleted, so there should be no memory loss.